src/EVPFFT/scripts/build-it.sh

```diff
@@ -0,0 +1,101 @@
+    echo "  --heffte_build_type=<fftw|cufft|rocfft>"
+    echo "  --kokkos_build_type=<serial|openmp|pthreads|cuda|hip>"
+    echo "  --build_cores=<Integers with inclusive range of 1-32>. This argument is optional, default is set to 1"
+    echo "  --help: Display this help message"
+    return 1
+}
+
+# Initialize variables with default values
+machine=""
+heffte_build_type=""
+kokkos_build_type=""
+build_cores="1"
+fullmachine=""
+fullheffte=""
+fullkokkos=""
+fullcores=""
+
+# Define arrays of valid options
+valid_machines=("darwin" "chicoma" "linux" "mac")
+valid_heffte_build_types=("fftw" "cufft" "rocfft")
+valid_kokkos_build_types=("serial" "openmp" "pthreads" "cuda" "hip")
+
+# Parse command line arguments
+for arg in "$@"; do
+    case "$arg" in
+        --machine=*)
+            fullmachine=$arg
+            option="${arg#*=}"
+            if [[ " ${valid_machines[*]} " == *" $option "* ]]; then
+                machine="$option"
+            else
+                echo "Error: Invalid --machine specified."
+                show_help
+                return 1
+            fi
+            ;;
+        --heffte_build_type=*)
+            fullheffte=$arg
+            option="${arg#*=}"
+            if [[ " ${valid_heffte_build_types[*]} " == *" $option "* ]]; then
+                heffte_build_type="$option"
+            else
+                echo "Error: Invalid --heffte_build_type specified."
+                show_help
+                return 1
+            fi
+            ;;
+        --kokkos_build_type=*)
+            fullkokkos=$arg
+            option="${arg#*=}"
+            if [[ " ${valid_kokkos_build_types[*]} " == *" $option "* ]]; then
+                kokkos_build_type="$option"
+            else
+                echo "Error: Invalid --kokkos_build_type specified."
+                show_help
+                return 1
+            fi
+            ;;
+        --build_cores=*)
+            fullcores=$arg
+            option="${arg#*=}"
+            if [ $option -ge 0 ] && [ $option -le 32 ]; then
+                build_cores="$option"
+            else
+                echo "Error: Invalid --build_cores specified."
+                show_help
+                return 1
+            fi
+            ;;
+        --help)
+            show_help
+            return 1
+            ;;
+        *)
+            echo "Error: Invalid argument or value specified."
+            show_help
+            return 1
+            ;;
+    esac
+done
+
+# Check if required options are specified
+if [ -z "$machine" ] || [ -z "$heffte_build_type" ] || [ -z "$kokkos_build_type" ]; then
+    echo "Error: --machine, heffte_build_type and --kokkos_build_type are required options."
+    show_help
+    return 1
+fi
+echo "Your options of $machine $heffte_build_type $kokkos_build_type are valid! Let's start building"
+
+cd "$( dirname "${BASH_SOURCE[0]}" )"
+
+source setup-env.sh ${fullmachine} ${fullheffte} ${fullkokkos} ${fullcores}
```

Comment:
here we can do
`source setup-env.sh --machine=$machine --heffte_build_type=$heffte_build_type etc..`